ngraph/core/reference/include/ngraph/runtime/reference/convolution_backprop_data.hpp

```diff
@@ -0,0 +1,309 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cfenv>
+#include <cmath>
+#include <functional>
+#include <numeric>
+
+#include "ngraph/axis_vector.hpp"
+#include "ngraph/coordinate_transform.hpp"
+#include "ngraph/runtime/reference/concat.hpp"
+#include "ngraph/runtime/reference/helpers.hpp"
+#include "ngraph/runtime/reference/reverse.hpp"
+#include "ngraph/runtime/reference/split.hpp"
+#include "ngraph/util.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            // in: NC_I...
+            // filter: C_OC_I...
+            // out: NC_O...
+            template <typename INPUT,
+                      typename FILTER,
+                      typename OUTPUT,
+                      typename ACCUMULATION = typename widen<OUTPUT>::type>
+            void convolution_backprop_impl(const INPUT* in,
+                                           const FILTER* filter,
+                                           OUTPUT* out,
+                                           const Shape& in_shape,
+                                           const Shape& filter_shape,
+                                           const Shape& out_shape,
+                                           const Strides& stride,
+                                           const Strides& filter_dilation,
+                                           const CoordinateDiff& in_pad_below,
+                                           const CoordinateDiff& in_pad_above,
+                                           const Strides& in_dilation,
+                                           size_t in_batch_axis,
+                                           size_t in_channel_axis,
+                                           size_t filter_out_channel_axis,
+                                           size_t filter_in_channel_axis,
+                                           size_t out_batch_axis,
+                                           size_t out_channel_axis)
+            {
+                auto old_mode = std::fegetround();
```

Comment:
ConvolutionBackpropData will be addressed in different ticket, so do not review it's code - I didn't change it, only moved it to other file because previously it was together with convolution.